applications/CoSimulationApplication/tests/test_rigid_body_solver.py

```diff
@@ -0,0 +1,2047 @@
+                            "mass": 100.0,
+                            "stiffness": 1000.0,
+                            "damping": 10.0
+                        }
+                    },
+                    {
+                        "dof": "rotation_x",
+                        "constrained": false,
+                        "system_parameters": {
+                            "mass": 2.0,
+                            "stiffness": 200.0,
+                            "damping": 2.0
+                        }
+                    },
+                    {
+                        "dof": "rotation_y",
+                        "constrained": false,
+                        "system_parameters": {
+                            "mass": 10.0,
+                            "stiffness": 1000.0,
+                            "damping": 10.0
+                        }
+                    },
+                    {
+                        "dof": "rotation_z",
+                        "constrained": false,
+                        "system_parameters": {
+                            "mass": 5.0,
+                            "stiffness": 500.0,
+                            "damping": 5.0
+                        }
+                    }
+                ]
+            }
+        }''')
+        Parameters.RecursivelyAddMissingParameters(self.default_parameters)
+
+        simulation = RigidBodySolver(self.model, Parameters)
+        
+        Constrains = {'displacement_x': False, 'displacement_y': False, 'displacement_z': False, 'rotation_x': False, 'rotation_y': False, 'rotation_z': False}
+        M = [[ 40.,  0. ,   0. ,  0.,   0.,   0.],
+            [  0. , 50. ,   0. ,  0. ,  0. ,  0.],
+            [  0. ,  0. , 100. ,  0. ,  0. ,  0.],
+            [  0. ,  0. ,   0. ,  2. ,  0. ,  0.],
+            [  0. ,  0. ,   0. ,  0. , 10. ,  0.],
+            [  0. ,  0. ,   0. ,  0. ,  0. ,  5.]]
+        C = [[ 4.,  0.,  0.,  0.,  0.,  0.],
+             [ 0.,  5.,  0.,  0.,  0.,  0.],
+             [ 0.,  0., 10.,  0.,  0.,  0.],
+             [ 0.,  0.,  0.,  2.,  0.,  0.],
+             [ 0.,  0.,  0.,  0., 10.,  0.],
+             [ 0.,  0.,  0.,  0.,  0.,  5.]]
+        K = [[ 400.,    0.,    0.,    0.,    0.,    0.],
+             [   0.,  500.,    0.,    0.,    0.,    0.],
+             [   0.,    0., 1000.,    0.,    0.,    0.],
+             [   0.,    0.,    0.,  200.,    0.,    0.],
+             [   0.,    0.,    0.,    0., 1000.,    0.],
+             [   0.,    0.,    0.,    0.,    0.,  500.]]
+
+        for i, dof in enumerate(['displacement_x', 'displacement_y', 'displacement_z', 'rotation_x', 'rotation_y', 'rotation_z']):
+            self.assertEqual(simulation.is_constrained[dof], Constrains[dof])
+            for j in range(6):
+                self.assertEqual(simulation.M[i][j], M[i][j])
+                self.assertEqual(simulation.C[i][j], C[i][j])
+                self.assertEqual(simulation.K[i][j], K[i][j])
```

Comment:
you can avoid the last for loop and use `self.assertMatrixAlmostEqual(m1,m2, precision)`

Reply:
I was unable to use assertMatrixAlmostEqual because the matrix that it evaluates uses numpy.ndarray and it doesn't have Size1 and Size2 in its attribute. But it is possible to use just one loop and use assertVectorAlmostEqual instead of assertMatrixAlmostEqual.

Reply:
In that case, I would keep it like this.